infra/gravity/cluster_resize.go

```diff
@@ -68,22 +68,24 @@ func (c *TestContext) ShrinkLeave(nodesToKeep, nodesToRemove []Gravity) error {
 	return trace.Wrap(utils.CollectErrors(ctx, errs))
 }
 
-// RemoveNode simulates sudden nodes loss within an existing cluster followed by node eviction
-func (c *TestContext) RemoveNode(nodesToKeep []Gravity, remove Gravity) error {
+// RemoveNodes evicts several nodes from the cluster
+func (c *TestContext) RemoveNodes(nodesToKeep, nodesToRemove []Gravity) error {
 	if len(nodesToKeep) == 0 {
 		return trace.BadParameter("node list empty")
 	}
 
-	master := nodesToKeep[0]
-
 	ctx, cancel := context.WithTimeout(c.ctx, c.timeouts.Leave)
 	defer cancel()
 
-	err := master.Remove(ctx, remove.Node().PrivateAddr(), Graceful(!remove.Offline()))
-	if err != nil {
-		return trace.Wrap(err)
+	master := nodesToKeep[0]
+
+	errs := make(chan error, len(nodesToRemove))
+	for _, node := range nodesToRemove {
+		go func(n Gravity) {
+			err := master.Remove(ctx, n.Node().PrivateAddr(), Graceful(!n.Offline()))
```

Comment:
Seems like it can now launch multiple "shrink" operations simultaneously? Gravity actually doesn't support that, so if there's an ongoing shrink operation, subsequent "gravity remove" commands will exit with error until the current one completes. I think that's the reason this only accepted a single node. It might be fine to keep this updated interface in robotest for future when (or if) we actually start allowing simultaneous shrinks (or tweak remove command so it waits till the current operation completes).

Reply:
Thanks! I didn't realize that.  I'll serialize this loop and update the timeouts accordingly.

Reply:
Reworked in 1ed898d.